### DISCUSSION TITLE
v5.0.0

### DISCUSSION THREAD

**Discussion by EtienneLem** [missive/emoji-mart#573] [Announcements]
# EmojiMart for the Web™
A new generation of **[🏪 EmojiMart](https://missiveapp.com/open/emoji-mart)** is finally upon us! Leveraging modern web APIs, it should be faster than ever and much more accessible. Both in terms of accessibility (a11y) and development/adoption. EmojiMart is now a custom HTML component and can be dropped in any website, whatever framework you’re using. The codebase itself is much more simpler than its predecessor, I hope it will encourage the community to participate as it always did. It will definitely be easier for us to maintain and address issues.

## Data
Data has been completely decoupled, which will give you more flexibility to control your app bundle size.

### Emoji 14.0
[Emoji 14.0](https://emojipedia.org/emoji-14.0) is the currently supported emoji version. EmojiMart will however try and detect the most recent version supported by your browser and will ignore unsupported emojis.

## Breaking changes, migrating and documentation
Depending on your usage, migrating should be relatively easy. Haven’t had time to work on advanced documentation yet, but [examples available here](https://github.com/missive/emoji-mart/tree/main/packages/emoji-mart-website) should be a good starting point.

Still some work left around developer experience (docs, typings, etc.), but the picker itself felt stable enough to release and gather feedback. [Discussions](https://github.com/missive/emoji-mart/discussions) have been enabled.

<hr /><em>This discussion was created from the release <a href='https://github.com/missive/emoji-mart/releases/tag/'>v5.0.0</a>.</em>

**Comment by romellem** [comment]
This is great, thanks for all your hard work on this!

Just curious, what happened to `v4.0.0`? I notice this went from `v3` to `v5` and was just wondering why `v4` was skipped?

Great work again, thanks!

**Reply by EtienneLem** [reply]
Hi 👋🏼, thanks for the kind words.

No particular reason; just to highlight the fact that it was a pretty major update. It deserved a nice *v5.0.0* 😎

**Comment by Leandro096** [comment]
Hello. I was just trying on this library but is giving me an error (maybe it would be because I am a junior developer and I could have something missing) I would like to know if anybody can help me.

I am using Tailwind CSS with Next.js and I tried to use this library but I have been trying to get everything done but the error is the following:

![WhatsApp Image 2022-04-26 at 6 28 25 PM](https://user-images.githubusercontent.com/67251206/165433281-feaa5e2d-5c28-4102-b41c-1d25720f8f58.jpeg)

Thanks for all your help.

**Reply by eremenko1** [reply]
I get the same thing, nextjs
![image](https://user-images.githubusercontent.com/67428314/165474126-ca317617-7d11-4317-95ca-130814039237.png)

**Reply by omridevk** [reply]
me to, window is undefined, I assume it is the access to localStorage that was added.

**Reply by mattcasey** [reply]
it's actually due to the HTML element added, the code is extending a class attached to window: https://github.com/missive/emoji-mart/blob/main/packages/emoji-mart/src/components/HTMLElement/HTMLElement.js#L1. In the localStorage utils, there's a try/catch around window access

**Comment by omridevk** [comment]
https://github.com/missive/emoji-mart/blob/5d04680c0503410752256ed1f744166ac48ee219/packages/emoji-mart/src/helpers/store.js

**Comment by mattcasey** [comment]
Not sure if this is the right place, but it looks like v5 still needs Typescript types and the DefinitelyTyped lib has not been updated yet, it's still on v3

**Reply by arochadaniel** [reply]
Had to downgrade into V3 due to missing types

**Reply by rugk** [reply]
This has been reported, but it's still the case as of 2026, unfortunately: https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/60094

**Comment by Ethorsen** [comment]
Do we still have access to getEmojiDataFromNative util function?
